### PR TITLE
searchapi: aggregator changes

### PR DESCRIPTION
**Pull request by vladbmihaescu**
* refactored code that displays nested aggregations

Screenshot from 2019-04-09 18-10-56

### CODE REVIEW

File: src/api/SearchApi.js [hunk 1]
```diff
@@ -74,39 +65,53 @@ class RequestSerializer {
...
-      key: bucket.key,
-      total: bucket.doc_count,
-      hasNestedField: false,
-    };
+const extractFromPath = (aggregation, pathToExtract) => {
+  return pathToExtract.reduce((obj, key) =>
+    (obj && obj[key] !== 'undefined') ? obj[key] : undefined, aggregation);
+}
+
+const createAggregation = (parentKeyName, buckets) => {
```
Comment (ntarocco) [inline src/api/SearchApi.js]: What do you think if you encapsulate both functions inside `ResponseSerializer` class?

File: src/api/SearchApi.js [hunk 2]
```diff
@@ -74,39 +65,53 @@ class RequestSerializer {
...
+      aggregations[item.key] = createAggregation(newKeyName, nestedBuckets, nestedAggregationPath);
+      aggregations[item.key]["key"] = newKeyName;
+      aggregations[item.key]["name"] = item.key;
+      aggregations[item.key]["total"] = item.doc_count;
+      aggregations[item.key]["hasNestedField"] = nestedField;
     }
-    return aggregation;
-  }
+    else {
+      delete item.subtype;
```
Comment (ntarocco) [inline src/api/SearchApi.js]: Does this `item.subtype` always exists?

File: src/api/SearchApi.js [hunk 3]
```diff
@@ -74,39 +65,53 @@ class RequestSerializer {
...
+      aggregations[item.key] = item;
```
Comment (ntarocco) [inline src/api/SearchApi.js]: I think you can delete this line no? Why not just:

```
const item = {
    'hasNestedField': false,
    'name': item.key,
    'total': item.doc_count,
    'key': parentKeyName.concat(".", item.key)
}
aggregations[item.key] = item;
```
But I might miss something here... the values maybe?

File: src/components/Aggregator/AggregatorValues.js [hunk 4]
```diff
@@ -20,6 +20,47 @@ export default class AggregatorValues extends Component {
...
+  buildAggregations = (fieldName, values, userSelection) => {
+    const parentNestedField = values.hasNestedField;
+
+    return Object.keys(values).map((key, index) => {
+      if (typeof values[key] === 'object') {
+        const aggrValue = values[key];
+        const checked = this.isItemChecked(fieldName, aggrValue, userSelection);
+        const _onUserSelectionChange = (e, { value }) => {
+          const aggrNameValue = {};
+          const aggTypeName = parentNestedField ? parentNestedField: fieldName; // assuming the invenio bug where sub-aggregations with the same name have one type of url request
```
Comment (ntarocco) [inline src/components/Aggregator/AggregatorValues.js]: You cannot assume which backend you are using, the UI doesn't know it :)
I would re-phrase to something like:
`assuming that root fields names are unique`?

File: src/api/SearchApi.js (hunk 2 above)
Comment (vladbmihaescu) [reply src/api/SearchApi.js]: I missed this, yes, now it deletes based on what the sub aggregation name is.

File: src/components/Aggregator/AggregatorValues.js (hunk 4 above)
Comment (vladbmihaescu) [reply src/components/Aggregator/AggregatorValues.js]: I should've removed the line, it's not used anymore, now the keys are unique so it's not needed.

File: src/api/SearchApi.js [hunk 5]
```diff
@@ -76,38 +76,54 @@ class RequestSerializer {
...
 /** Default backend response serializer */
 class ResponseSerializer {
-  _serializeAggregation(bucket) {
-    const aggregation = {
-      key: bucket.key,
-      total: bucket.doc_count,
-      hasNestedField: false,
-    };
+  _serializeAggregations(aggregationsResponse) {
+      const createAggregation = (parentKeyName, buckets) => {
```
Comment (zzacharo) [inline src/api/SearchApi.js]: I think split that to a `_ createAggregation` function would be more readable... WDYT?

File: src/api/SearchApi.js [hunk 6]
```diff
@@ -76,38 +76,54 @@ class RequestSerializer {
...
+              "name": item.key,
+              "total": item.doc_count,
+              "key": "".concat(parentKeyName, ".", item.key)
+          };
+        }
+      });
+      return aggregations;
+    }
+
+    const extractFromPath = (aggregation, pathToExtract) => {
```
Comment (zzacharo) [inline src/api/SearchApi.js]: Maybe this one also?

File: src/api/SearchApi.js (hunk 5 above)
Comment (vladbmihaescu) [reply src/api/SearchApi.js]: I agree, done!

File: demo/src/index.js [hunk 7]
```diff
@@ -56,7 +92,7 @@ class Demo extends Component {
       <div>
         <ReactSearchKit
           searchConfig={{
-            url: 'https://zenodo.org/api/records/',
+            url: 'https://zenodo.org/api/records/',//,'https://localhost:5000/api/documents/'
```
Comment (ntarocco) [inline demo/src/index.js]: @vladbmihaescu can you please remove this comment?

Comment (vladbmihaescu) [reply demo/src/index.js]: Done!

**Pull request merged**